fwdpy11/headers/fwdpy11/discrete_demography/ExponentialGrowth.hpp

```diff
@@ -0,0 +1,66 @@
+#ifndef FWDPY11_EXPONENTIAL_GROWTH_HPP
+#define FWDPY11_EXPONENTIAL_GROWTH_HPP
+
+#include <cmath>
+#include <stdexcept>
+#include "GrowthRate.hpp"
+
+namespace fwdpy11
+{
+    struct ExponentialGrowth : public GrowthRate
+    {
+        ExponentialGrowth(std::uint32_t o, std::uint32_t n0, double r)
+            : GrowthRate(o, n0, r)
+        {
+            if (G <= 0.0)
+                {
+                    throw std::invalid_argument("growth rate must be > 0");
+                }
+            if (!std::isfinite(G))
+                {
+                    throw std::invalid_argument("growth rate must be finite");
+                }
+        }
+
+        std::uint32_t
+        operator()(const std::uint32_t current_generation) const override final
+        {
+            //if(G == 0.0) { return N0; }
+            return std::round(static_cast<double>(N0)
+                              * std::pow(G, static_cast<double>(
+                                                current_generation - onset)));
+        }
+
+        std::unique_ptr<GrowthRate>
+        clone() const final
+        {
+            return std::unique_ptr<ExponentialGrowth>(
+                new ExponentialGrowth(*this));
+        }
+
+        void
+        reset_size(std::uint32_t when, std::uint32_t new_size) override final
```

Comment:
Need to add an option to change the behaviour of this class, and only reset N0 and the time, but not the rate.  Thus, the growth of the deme would be "persistent" through external changes in sizes.

Reply:
Resolved via https://github.com/molpopgen/fwdpy11/pull/237/commits/e407b0c30e064126d34a870bc580b8d27eefdb3f